recipes/libmodbus/all/conanfile.py

```diff
@@ -23,101 +29,116 @@ class LibmodbusConan(ConanFile):
-
-    @property
-    def _settings_build(self):
-        return self.settings_build if hasattr(self, "settings_build") else self.settings
-
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            try:
+                del self.options.fPIC # once removed by config_options, need try..except for a second del
+            except Exception:
+                pass
+        try:
+            del self.settings.compiler.libcxx # for plain C projects only
+        except Exception:
+            pass
+        try:
+            del self.settings.compiler.cppstd # for plain C projects only
+        except Exception:
+            pass
 
     def build_requirements(self):
-        if self.settings.compiler == "Visual Studio":
-            self.build_requires("automake/1.16.3")
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
+        if self.settings.os == "Macos" or self.settings.compiler == "Visual Studio":
+            self.tool_requires("automake/1.16.3")
+        # see https://github.com/conan-io/conan/issues/11969
+        bash_path = os.getenv("CONAN_BASH_PATH") or self.conf.get("tools.microsoft.bash:path")
+        if self.settings.os == "Windows" and not bash_path:
+            self.tool_requires("msys2/cci.latest")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
-
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
-        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-        conf_args = [
-            "--disable-tests",
-            "--without-documentation",
-            "--prefix={}".format(tools.unix_path(self.package_folder)),
-        ]
-        if self.options.shared:
-            conf_args.extend(["--enable-shared", "--disable-static"])
-        else:
-            conf_args.extend(["--enable-static", "--disable-shared"])
-        if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) >= "12":
+        get(self, **self.conan_data["sources"][self.version],
+                destination=self.source_folder, strip_root=True)
+
+    def validate(self):
+        # validate the minimum cpp standard supported
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, 11)
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def generate(self):
+        # --fpic is automatically managed when 'fPIC' option is declared
+        # --enable/disable-shared is automatically managed when 'shared' option is declared
```

Comment:
```suggestion
```
Not needed. The documentation is very clear and it's expected that everyone read it.